files/cron.go

```diff
@@ -0,0 +1,341 @@
+package files
+
+// Shamelessly lifted from https://github.com/temporalio/temporal/blob/main/service/worker/scheduler/calendar.go
+// I'd import it as a dependency, but it's not exported from the package, so this is the
+// only way to use it from here.
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	schedpb "go.temporal.io/api/schedule/v1"
+	"go.temporal.io/server/common/primitives/timestamp"
+	"google.golang.org/protobuf/types/known/durationpb"
+)
+
+type parseMode int
+
+const (
+	// minCalendarYear is the smallest year that can appear in a calendar spec.
+	minCalendarYear = 2000
+	// maxCalendarYear is the latest year that will be recognized for calendar dates.
+	// If you're still using Temporal in 2100 please change this constant and rebuild.
+	maxCalendarYear = 2100
+)
+
+const (
+	// Modes for parsing range strings: all modes accept decimal integers.
+	parseModeInt parseMode = iota
+	// parseModeYear is like parseModeInt but returns an empty range for the default.
+	parseModeYear
+	// parseModeMonth also accepts month name prefixes (at least three letters).
+	parseModeMonth
+	// parseModeDow also accepts day-of-week prefixes (at least two letters).
+	parseModeDow
+)
+
+var (
+	errOutOfRange = errors.New("out of range")
+
+	monthStrings = []string{ //nolint:gochecknoglobals
+		"january",
+		"february",
+		"march",
+		"april",
+		"may",
+		"june",
+		"july",
+		"august",
+		"september",
+		"october",
+		"november",
+		"december",
+	}
+
+	dowStrings = []string{ //nolint:gochecknoglobals
+		"sunday",
+		"monday",
+		"tuesday",
+		"wednesday",
+		"thursday",
+		"friday",
+		"saturday",
+	}
+)
+
+func parseCronString(cronStr string) (*schedpb.StructuredCalendarSpec, *schedpb.IntervalSpec, string, error) {
```

Comment:
Can we use the same validation logic as we do in the server? https://github.com/amp-labs/server/blob/6881043287fe5c8307b92d6e3ea6f1607c501f5f/shared/utils/validate.go#L53 

Or as an easier solution for now, the CLI can just check that the `schedule` string exists, and then leave it to the server to validate syntax and frequency.